### PR TITLE
Allow authenticating with username and email

### LINKED ISSUE #473: Allow username or email for login
When a user registers for NervesHub, they create a username and an email address with their account.

The username basically never gets used, mentioned, or displayed again in the web experience and login is based on email address. In the CLI, username is used a little more frequently, but not much.

If we're requiring both, we should allow the user to login with either username or email address

### PR DESCRIPTION
**Pull request by hassanshaikley**
fixes #473 

💚

### CODE REVIEW

**Comment by hassanshaikley** [comment]
If we look for username or email in `nerves_hub_web_core/accounts.ex` then we shouldn't need to adjust the form, right?

I'll get on these changes!

**Comment by jjcarstens** [comment]
I think the form will need to be adjusted somewhat. Because right now it uses `email_tag` which attempts to validate the field before submitting:

image

And we'll want to change text to indicate `username` or `email` could be used prob.

**Comment by hassanshaikley** [comment]
I see! Awesome.

**Comment by hassanshaikley** [comment]
Alright, just pushed.

Some things I am not sure about:

1. Do I still match on username for the CLI? Or should I remove that code as it will work if you refer to the username parameter as "email" anyways.
2. Is it acceptable to keep calling it email on the front end or should I change it to `email_or_username`?

Thanks!

**Comment by jjcarstens** [comment]
1. I think its fine to leave that match in for the CLI. Then they have the option to be more explicit or not
2. `email_or_username` might be a little more correct, but I'm good either way

File: apps/nerves_hub_www/lib/nerves_hub_www_web/templates/session/new.html.eex [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <div class="modal-body">
       <div class="form-group">
         <label for="email_input">Email</label>
```
Comment (jjcarstens) [inline apps/nerves_hub_www/lib/nerves_hub_www_web/templates/session/new.html.eex]: I think we should change this label to `Email/Username` (or something of the sort)

**Comment by hassanshaikley** [comment]
How's that!

**Comment by hassanshaikley** [comment]
😄

**Comment by mobileoverlord** [comment]
This has been merged into master manually, Thanks!

**Pull request closed without merging**